packages/joy-media/src/Upload.tsx

```diff
@@ -263,20 +271,19 @@ class Component extends React.PureComponent<Props, State> {
 
     // TODO: validate url .. must start with http
 
-    this.setState({ discovering: false, uploading: true });
+    this.setState({ discovering: false, progress: 0, uploading: true });
```

Comment:
It just feels like easier to read: first we write two states: discovering, uploading; 
then we write 0 to progress.
```suggestion
    this.setState({ discovering: false, uploading: true, progress: 0 });
```